app/src/main/java/com/codesoom/assignment/DemoHandler.java

```diff
@@ -0,0 +1,173 @@
+package com.codesoom.assignment;
+
+import com.codesoom.assignment.models.HttpResponse;
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.models.TaskList;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.*;
+import java.net.URI;
+import java.util.stream.Collectors;
+
+public class DemoHandler implements HttpHandler {
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    TaskList taskList = new TaskList();
+
+    public static final String slash = "/";
+    public static final String GET = "GET";
+
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+
+        String[] pathArray = createPathArray(exchange);
+
+        HttpResponse httpResponse = new HttpResponse("", 200);
+        try {
+            if (isTasksRequest(pathArray)) {
+                httpResponse = fetchHttpResponse(exchange);
+            }
+        } catch (NullPointerException e) {
+            httpResponse = new HttpResponse("", 404);
+        } finally {
+            sendResponse(exchange, httpResponse);
+        }
+    }
+
+    private int getRequestTaskId(String[] pathArray) {
+        String taskIdStr = pathArray[2];
+        return Integer.parseInt(taskIdStr);
+    }
+
+    private String[] createPathArray(HttpExchange exchange) {
+        URI uri = exchange.getRequestURI();
+        String path = uri.getPath();
+        return path.split(slash);
+    }
+
+
+    private boolean isTasksRequest(String[] pathArray) {
+        return pathArray.length >= 2 && pathArray[1].equals("tasks");
+    }
+
+    private HttpResponse fetchHttpResponse(HttpExchange exchange) throws IOException {
+        String requestMethod = exchange.getRequestMethod();
+        String[] pathArray = createPathArray(exchange);
+
+        HttpResponse httpResponse;
+        switch (requestMethod) {
+            case "GET":
+                httpResponse = fetchTask(pathArray);
+                break;
+            case "POST":
+                httpResponse = insertTask(createBody(exchange));
+                break;
+            case "PUT":
+                httpResponse = updateTask(pathArray, createBody(exchange));
+                break;
+            default:
+                httpResponse = deleteTask(pathArray);
```

Comment:
default는 예외인 경우에 쓰는 것으로, delete를 예외로 쓰시면 안됩니다. 정상적으로 사용한다면 이렇게 써야돼요
```suggestion
            case "PUT":
                httpResponse = updateTask(pathArray, createBody(exchange));
                break;
            case "DELETE":
                httpResponse = deleteTask(pathArray);
                break;
            default:
                throw new UnsupportedMethod("지원하지 않는 메서드 입니다. 메서드: " + requestMethod);
```

Reply:
예외처리를 못했던 케이스였네요!
default는 윤석님께서 말씀해주신 UnsupportedMethod 를 던지도록 구현했습니다!